connectivity/tests/from-cidr.go

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package tests
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cilium/cilium-cli/connectivity/check"
+)
+
+// FromCIDRToPod generates HTTP request from each node without Cilium to the
+// echo pods within the Cilium / K8s cluster.
+func FromCIDRToPod() check.Scenario {
+	return &fromCIDRToPod{}
+}
+
+// fromCIDRToPod implements a Scenario.
+type fromCIDRToPod struct{}
+
+func (f *fromCIDRToPod) Name() string {
+	return "from-cidr-to-pod"
+}
+
+func (f *fromCIDRToPod) Run(ctx context.Context, t *check.Test) {
+	clientPod := t.Context().HostNetNSPodsByNode()[t.NodesWithoutCilium()[0]]
+	i := 0
+
+	for _, pod := range t.Context().EchoPods() {
+		ep := check.HTTPEndpoint(
+			"http-endpoint",
+			fmt.Sprintf("%s://%s:%d%s", pod.Scheme(), pod.Address(check.IPFamilyAny), pod.Port(), pod.Path()),
```

Comment:
will not work for IPv6 because of the `:`, use https://pkg.go.dev/net#JoinHostPort